op-node/rollup/engine/engine_controller.go

```diff
@@ -567,3 +580,214 @@ func (e *EngineController) TryBackupUnsafeReorg(ctx context.Context) (bool, erro
 	return true, derive.NewTemporaryError(fmt.Errorf("cannot restore unsafe chain using backupUnsafe: err: %w",
 		eth.ForkchoiceUpdateErr(fcRes.PayloadStatus)))
 }
+
+// OnEvent implements event.Deriver (moved from EngDeriver)
+func (d *EngineController) OnEvent(ctx context.Context, ev event.Event) bool {
+	d.mu.Lock()
+	defer d.mu.Unlock()
+	switch x := ev.(type) {
```

Comment:
Can we preserve the comment:
```
    // TODO(#16917) Remove Event System Refactor Comments
	//  PromoteUnsafeEvent, PromotePendingSafeEvent, PromoteLocalSafeEvent fan out is updated to procedural 
```
My understanding is that the onEvent Handler originally from `EngDeriver` is embedded into `EngineController` now.